src/GraphQL.Conventions/Types/Resolution/DependencyInjectorExtensions.cs

```diff
@@ -1,10 +1,17 @@
-using System.Reflection;
+using System;
 
 // ReSharper disable once CheckNamespace
 namespace GraphQL.Conventions
 {
-    public static class DependencyInjectorExtensions
+    internal static class DependencyInjectorExtensions
     {
-        public static T Resolve<T>(this IDependencyInjector dependencyInjector) => (T)dependencyInjector?.Resolve(typeof(T).GetTypeInfo());
+        public static T GetService<T>(this IServiceProvider serviceProvider) => (T)serviceProvider?.GetService(typeof(T));
+
+        public static object GetRequiredService(this IServiceProvider serviceProvider, Type serviceType)
+            => serviceProvider.GetService(serviceType)
+            ?? throw new InvalidOperationException($"No service for type '{serviceType}' has been registered.");
```

Comment:
In main repo we use `GetFriendlyName`. Copy here?

Reply:
I can incorporate this in follow-up mods that I'll be making - thanks